windows-agent/cmd/ubuntu-pro-agent/agent/agent_test.go

```diff
@@ -508,3 +509,7 @@ func captureStdout(t *testing.T) func() string {
 		return out.String()
 	}
 }
+
+func TestWithWslSystemMock(t *testing.T) {
```

Comment:
I need the function `testutils.MockWslSystemCmd()` in the binary when testing the agent. I guess I currently can't have it "automatically" because that's the only function in `testutils` and thus, if I don't explicitly call it, then it's an unused package. If instead, I have an API to override the defaults and call it, then I get that function in binary transparently? Let me test that.

Reply:
Couldn't figure out a clean way to allow invoking the current binary as the mock for the subprocess (`wsl --system` in this case).